src/Runner.Worker/ActionManager.cs

```diff
@@ -503,39 +503,44 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
                 string apiUrl = GetApiUrl(executionContext);
                 string archiveLink = BuildLinkToActionArchive(apiUrl, repositoryReference.Name, repositoryReference.Ref);
                 Trace.Info($"Download archive '{archiveLink}' to '{destDirectory}'.");
-                await DownloadRepositoryActionAsync(executionContext, archiveLink, destDirectory);
+                var downloadDetails = new ActionDownloadDetails(archiveLink, ConfigureAuthorizationFromContext);
+                await DownloadRepositoryActionAsync(executionContext, downloadDetails, destDirectory);
                 return;
             }
             else
             {
                 string apiUrl = GetApiUrl(executionContext);
 
                 // URLs to try:
-                var archiveLinks = new List<string> {
+                var downloadAttempts = new List<ActionDownloadDetails> {
                     // A built-in action or an action the user has created, on their GHES instance
                     // Example:  https://my-ghes/api/v3/repos/my-org/my-action/tarball/v1
-                    BuildLinkToActionArchive(apiUrl, repositoryReference.Name, repositoryReference.Ref),
-
-                    // A community action, synced to their GHES instance
-                    // Example:  https://my-ghes/api/v3/repos/actions-community/some-org-some-action/tarball/v1
-                    BuildLinkToActionArchive(apiUrl, $"actions-community/{repositoryReference.Name.Replace("/", "-")}", repositoryReference.Ref)
+                    new ActionDownloadDetails(
```

Comment:
Since each URL has its own auth method, I grouped them together into an object and pass that into the download method.  This is a little setup for https://github.com/github/c2c-actions-experience/issues/2727